realbook/layers/signal.py

```diff
@@ -380,3 +380,151 @@ def get_config(self) -> Dict[str, Any]:
             }
         )
         return config
+    
+
+class Magnitude(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            name: Optional[str] = None,
+            dtype: tf.dtypes.DType = tf.float32,
+            *args: Any,
+            **kwargs: Any,
+    ):
+        """
+        A Tensorflow Keras layer that calculates the magnitude of a complex tensor.
+        The input is complex-valued with shape (num_batches, time, fft_length // 2 + 1, 2).
+        The output is real-valued with shape (num_batches, time, fft_length // 2 + 1)
+
+        Args:
+            name: Name of the layer.
+            dtype: Type used in calculation.
+        """
+        super().__init__(name=name, dtype=dtype, *args, **kwargs)
+
+    def call(self, inputs: tf.Tensor) -> tf.Tensor:
+        return tf.math.abs(inputs)
+
+    def get_config(self) -> Dict[str, Any]:
+        config: Dict[str, Any] = super().get_config().copy()
+        return config
+
+
+class Phase(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            name: Optional[str] = None,
+            dtype: tf.dtypes.DType = tf.float32,
+            *args: Any,
+            **kwargs: Any,
+    ):
+        """
+        A Tensorflow Keras layer that calculates the phase of a complex tensor.
+        The input is complex-valued with shape (num_batches, time, fft_length // 2 + 1, 2).
+        The output is real-valued with shape (num_batches, time, fft_length // 2 + 1)
+
+        Args:
+            name: Name of the layer.
+            dtype: Type used in calculation.
+        """
+        super().__init__(name=name, dtype=dtype, *args, **kwargs)
+
+    def call(self, inputs: tf.Tensor) -> tf.Tensor:
+        return tf.math.angle(inputs)
+
+    def get_config(self) -> Dict[str, Any]:
+        config: Dict[str, Any] = super().get_config().copy()
+        return config
+
+
+class MagnitudeToDecibel(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            ref: float = 1.0,
+            amin: float = 1e-10,
+            top_db: Optional[float] = None,
```

Comment:
```suggestion
            min_magnitude: float = 1e-10,
            top_db: Optional[float] = 80.0,
```
Correct me if I'm wrong, but I assume you based this layer on librosa's [power_to_db](https://librosa.org/doc/main/generated/librosa.power_to_db.html)? If so, I'd appreciate it if the defaults match librosa and would also appreciate it if the terminology is consistent within realbook itself. Although librosa doesn't directly state it,`amin` most likely means `minimum amplitude`. Since we are referring to amplitude as `magnitude` in realbook, I'd like amin to refer to magnitude and not amplitude.

It's fine to keep `top_db` as an `Optional`, but can you add the [conditional](https://github.com/librosa/librosa/blob/c35c21c21d366f9c30c674890f94b98d771256df/librosa/core/spectrum.py#L1783) that librosa has? Currently,`top_db` as an `Optional` doesn't make a difference in implementation and at worst will cause an exception to be raised since you're shouldn't be able to subtract a tensor from `None`.

Reply:
Yes, it should be similar to librosa's methods.
I have modified the code to match the values, and removed the Optional typing option.

Reply:
Could you please clarify what you mean when saying

> Since we are referring to amplitude as magnitude in realbook, I'd like amin to refer to magnitude and not amplitude.

Reply:
Let's ignore this conversation. It's all good as is.